[T Neethesh] iP

Neethesh26 opened the pull request:
### DukePro

> "Your mind is for having ideas, not holding them." - David Allen [(source)](https://dansilvestre.com/productivity-quotes)

DukePro frees your mind of having to remember things you need to do. It's, 

- text-based
- easy to learn
- ~FAST~ _SUPER_ FAST to use

All you need to do is, 

1. download it from [here](https://nus-cs2103-ay2223s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features: 

- [X] Managing tasks
- [ ] Managing deadlines (coming soon!)
- [ ] Reminders (coming soon)

-------------------------------------------------------------------------------------------

If you are a Java Programmer, you can use it to practice Java too. Here's the `main` method: 

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Neethesh26 committed:
Level-1

Neethesh26 committed:
Level-1

Neethesh26 committed:
Level-3

Neethesh26 committed:
Level-4

Neethesh26 committed:
Level-5

Neethesh26 committed:
Level-5

Neethesh26 committed:
Level-6

Neethesh26 committed:
Level-7 updates, included saving feature and updated task classes.

Neethesh26 committed:
Level-8 changes, added ability of event and deadline tasks to save date time information.

Neethesh26 committed:
Merge branch 'branch-Level-8'

Resolved conflicts
#	src/main/java/Deadline.java
#	src/main/java/Duke.java
#	src/main/java/DukeException.java
#	src/main/java/Task.java

Neethesh26 committed:
update dukeInfo.txt

Neethesh26 committed:
Merge branch 'add-gradle-support' of https://github.com/nus-cs2103-AY2223S1/ip

Neethesh26 committed:
Merge branch 'branch-Level-7'

Neethesh26 committed:
Added more oop  and organised classes into packages

Neethesh26 committed:
Created Junit test for Todo class and Deadline class

Neethesh26 committed:
Updating with the jar file created

Neethesh26 committed:
Update mainClassName in build.gradle file

Neethesh26 committed:
Created the javadocs and changed code quality to follow coding standards.

Neethesh26 committed:
Added find functionality.

Neethesh26 committed:
Updated code to adhere to coding standards

bryanljx commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,82 @@
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+public class Deadline extends Task{
```
Hmm, is there any reason to not provide header comments for public classes? Also, it's easy to miss out spaces like this case `Task{`, you might want to set up an actual code formatter or say intellij to auto format on save to avoid these 😄

bryanljx commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,82 @@
+
+public class Deadline extends Task{
+
+    /**
+     * Information of the date of deadline.
+     */
+    private LocalDate date;
```
Actually there isn't a really need to write javadocs for private fields but good that you have 👍

bryanljx commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,52 @@
+    }
+
+
+    public static void main(String[] args) {
+        new Duke("./data/dukeInfo.txt").run();
+
+    }
```
Is there a reason for the empty line on line 50? It looks kind of weird to me to randomly have an empty line before the closing brace 😅

bryanljx commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,111 @@
+    public static Command parse (String input) throws DukeException{
+        String[] arr = input.split(" ", 2);
+
+        if (input.equals("bye")) {
+            return new ExitCommand();
+        }
+
+        else if (input.equals("list")) {
+            return new ListCommand();
+        }
+
+        else if (arr[0].equals("mark")){
+            if (!inputChecker(arr)) {
+                throw DukeException.MarkIndexEmptyException();
+            } else {
+                return new MarkCommand(arr);
+            }
+
+        }
+        else if (arr[0].equals("unmark")) {
```
I kind of understand why the if else blocks were formatted this way because of how many else blocks there are though this is still a violation of the coding standard as shown below. Perhaps a switch block would be better here?

```java
if (condition) {
    statements;
} else if (condition) {
    statements;
} else {
    statements;
}
```

bryanljx commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,131 @@
+                    }
+                    tasks.add(toAdd);
+                    break;
+                default:
+
+                }
```
An empty default case feels a bit redundant here. Would adding a comment to explain why there are no other cases/this default case be reached or simply to remove this be better?

bryanljx commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,56 @@
+    /**
+     * Information of task completion.
+     */
+    private boolean completed;
```
Might be more appropriate here to name `completed` as `isCompleted` to be consistent with the coding standards defined for CS2103T though there are certainly cases where I would prefer `completed` to `isCompleted`

bryanljx commented on src/main/java/duke/command/DeadlineCommand.java:
```diff
@@ -0,0 +1,105 @@
+    private static void dateSetter(Task t, String date) throws DukeException{
+        String[] dateTime = date.split(" ", 2);
+
+        String ddmmyyyyRegex = "[0-9]{1,2}/[0-9]{1,2}/[0-9]{4}";
+        String yyyymmddRegex = "[0-9]{4}-[0-9]{1,2}-[0-9]{1,2}";
+        String timeRegex = "[0-9]{4}";
```
Sick! Nice use of regular expressions here 👍

bryanljx reviewed (commented):
Overall well written and compliant with the coding standard set in CS2103T. 

However, there are a few areas where there are redundant empty lines inserted which looked quite weird and spacing and line breaks could be more consistent as well. These are quite common mistakes that we will make so I would suggest actually setting up a code formatter or IntelliJ's auto format on save.

Keep up the good work!

Neethesh26 committed:
Installed javaFx dependencies

Neethesh26 committed:
Added javaFx

Neethesh26 committed:
Added assertions to test assumptions in code.

Neethesh26 committed:
Merge pull request #1 from Neethesh26/branch-Assertions

Added assertions to test assumptions in code.

Neethesh26 committed:
Improved code quality in alignment to standards set in textbook.

Neethesh26 committed:
Merge pull request #2 from Neethesh26/branch-CodeQuality

Improved code quality in alignment to standards set in textbook.

Neethesh26 committed:
Added new archive feature to dukebot

Currently, dukebot was is only able to delete tasks without saving.

This caused an issue to users who want to remove tasks from main list without deleting it full.

Now with the archive command, users can archive the whole list or specific tasks from the main list to the archive list for reference in the future.

Neethesh26 committed:
Merge pull request #4 from Neethesh26/branch-C-Archive

Added new archive feature to dukebot

Neethesh26 committed:
Error found in find command rectified

Neethesh26 committed:
Improved errror handling for user inputs that may occur during the run of duke.

Neethesh26 committed:
Updated the user guide and included the Ui.png file

Neethesh26 committed:
Reformat code
